src/DotVVM.Framework/ViewModel/Serialization/ViewModelSerializationMapper.cs

```diff
@@ -47,6 +47,7 @@ protected virtual ViewModelSerializationMap CreateMap(Type type)
         /// </summary>
         protected virtual IEnumerable<ViewModelPropertyMap> GetProperties(Type type)
         {
+            var isSetterSupported = type.IsGenericType && type.GetGenericTypeDefinition() == typeof(KeyValuePair<,>);
```

Comment:
I'd rather extract this as a separate method that would get a specific property - this decision is not made on the type level but on a property level.
Also, we should make this method virtual so it can be overridden if needed.